src/js/components/Select/index.d.ts

```diff
@@ -8,14 +8,27 @@ import {
   PlaceHolderType,
 } from '../../utils';
 
+interface onChangeEvent extends React.ChangeEvent<HTMLSelectElement> {
+  value: any;
+  option: any;
+  selected: number[];
+}
+
 export interface SelectProps {
   a11yTitle?: A11yTitleType;
   alignSelf?: AlignSelfType;
   gridArea?: GridAreaType;
-  children?: (...args: any[]) => any;
+  children?: (
+    option: object,
+    index: number,
+    options: object[],
+    state: { active: boolean; disabled: boolean; selected: boolean },
+  ) => React.ReactNode;
```

Comment:
I'm not sure this is completely accurate. Why not just remove the children from the Interface and fall back to React children type? 
From what you have specified, all the fields are required and not optional, which may cause backward compatibility issues for users who don't necessarily use them. 
i.e.
```suggestion
  children?: (
    option?: object,
    index?: number,
    options?: object[],
    state?: { active: boolean; disabled: boolean; selected: boolean },
  ) => React.ReactNode;
```

I think we should remove the `children` prop from the interface unless you think otherwise.